MLS.Agent/DotnetKernelJupyterInstaller.cs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.IO;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+using MLS.Agent.Tools;
+
+namespace MLS.Agent
+{
+    public static class DotnetKernelJupyterInstaller
+    {
+        public delegate Task<CommandLineResult> ExecuteCommand(string command, string args);
+
+        public static async Task<int> InstallKernel(ExecuteCommand executeCommand, IConsole console)
+        {
+            try
+            {
+                var result = await executeCommand("jupyter", "--paths");
+                if (result.ExitCode == 0)
+                {
+                    var dataDirectories = await GetJupyterDataPaths(result.Output.ToArray());
+                    Installkernel(dataDirectories, console);
+                    console.Out.WriteLine(".NET kernel installation succeded");
+                    return 0;
+                }
+                else
+                {
+                    throw new KernelInstallationFailureException($"Tried to invoke \"jupyter --paths\" but failed with exception: {string.Join("\n", result.Error)}");
+                }
+            }
+            catch (KernelInstallationFailureException e)
+            {
+                console.Out.WriteLine($".NET Kernel Installation failed with error {string.Join("\n", e.Message)}");
+                return -1;
+            }
+        }
+
+        private static void Installkernel(IEnumerable<DirectoryInfo> dataDirectories, IConsole console)
+        {
+            foreach (var directory in dataDirectories)
+            {
+                if (directory.Exists)
+                {
+                    var kernelDirectory = directory.Subdirectory("kernels");
+                    if (!kernelDirectory.Exists)
+                    {
+                        kernelDirectory.Create();
+                    }
+
+                    var dotnetkernelDir = kernelDirectory.Subdirectory(".NET");
+                    if (!dotnetkernelDir.Exists)
+                    {
+                        dotnetkernelDir.Create();
+                    }
+
+                    console.Out.WriteLine($"Installing the .NET kernel in directory: {dotnetkernelDir.FullName}");
+
+                    // Copy the files into the kernels directory
+                    File.Copy("kernels.json", Path.Combine(dotnetkernelDir.FullName, "kernels.json"));
```

Comment:
Is the working directory expected to contain these files? 

These will need to be extracted from our packaged tool like we do with other resources.